src/components/FileTreeNode/index.spec.tsx

```diff
@@ -216,4 +246,200 @@ describe(__filename, () => {
 
     expect(root.find(`.${styles.selected}`)).toHaveLength(0);
   });
+
+  it.each([
+    ['error', `.${styles.hasLinterErrors}`, 'times-circle'],
+    ['notice', `.${styles.hasLinterMessages}`, 'info-circle'],
+    ['warning', `.${styles.hasLinterWarnings}`, 'exclamation-triangle'],
+  ])('renders a file node that has linter %ss', (type, className, icon) => {
+    const message = {
+      ...fakeExternalLinterMessage,
+      file: 'manifest.json',
+      type: type as ExternalLinterMessage['type'],
+    };
+
+    const root = renderWithLinterMessage({
+      message,
+      treefoldRenderProps: {
+        isFolder: false,
+      },
+    });
+
+    expect(root.find(className)).toHaveLength(1);
+
+    const nodeIcons = root.find(`.${styles.nodeIcons}`);
+    expect(nodeIcons).toHaveLength(1);
+    expect(nodeIcons.find(FontAwesomeIcon)).toHaveProp('icon', icon);
+    expect(nodeIcons.find(FontAwesomeIcon).prop('title')).toMatch(
+      new RegExp(`file contains linter ${type}s`),
+    );
+  });
+
+  it.each([
+    ['error', `.${styles.hasLinterErrors}`, 'times-circle'],
+    ['notice', `.${styles.hasLinterMessages}`, 'info-circle'],
+    ['warning', `.${styles.hasLinterWarnings}`, 'exclamation-triangle'],
+  ])(
+    'renders a directory node that has linter %ss',
+    (type, className, icon) => {
+      const message = {
+        ...fakeExternalLinterMessage,
+        file: 'manifest.json',
+        type: type as ExternalLinterMessage['type'],
+      };
+
+      const root = renderWithLinterMessage({
+        message,
+        treefoldRenderProps: {
+          isFolder: true,
+        },
+      });
+
+      expect(root.find(className)).toHaveLength(1);
+
+      const nodeIcons = root.find(`.${styles.nodeIcons}`);
+      expect(nodeIcons).toHaveLength(1);
+      expect(nodeIcons.find(FontAwesomeIcon)).toHaveProp('icon', icon);
+      expect(nodeIcons.find(FontAwesomeIcon).prop('title')).toMatch(
+        new RegExp(`contains files with linter ${type}s`),
+      );
+    },
+  );
+
+  it('ignores linter messages unrelated to the current node', () => {
+    const message = {
+      ...fakeExternalLinterMessage,
+      file: 'manifest.json',
+    };
+
+    const root = renderWithLinterMessage({
+      message,
+      treefoldRenderProps: {
+        id: 'some/other/file.js',
+      },
+    });
+
+    expect(root.find(`.${styles.hasLinterMessages}`)).toHaveLength(0);
```

Comment:
I won't comment on them all but `.toHaveClassName()` would produce better messages in all of the assertions I'm seeing.

Reply:
`toHaveClassName()` and `.find()` are different. One is looking for a class name on the current wrapper, the other is used to make sure a component identified by its class name exits.